extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/SyncHttpClientConfig.java

```diff
@@ -0,0 +1,211 @@
+package io.quarkus.dynamodb.runtime;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class SyncHttpClientConfig {
+
+    /**
+     * Type of the sync HTTP client implementation
+     */
+    @ConfigItem(defaultValue = "url")
+    public SyncClientType type;
+
+    /**
+     * The maximum amount of time to establish a connection before timing out.
+     *
+     * <p>
+     * Used by `URL` and `APACHE` clients
+     */
+    @ConfigItem(defaultValue = "2S")
+    public Duration connectionTimeout;
+
+    /**
+     * The amount of time to wait for data to be transferred over an established, open connection before the connection is
+     * timed
+     * out.
+     * 
+     * <p>
+     * Used by `URL` and `APACHE` clients
+     */
+    @ConfigItem(defaultValue = "30S")
+    public Duration socketTimeout;
+
+    /**
+     * Apache HTTP client additional configuration
+     */
+    @ConfigItem(name = ConfigItem.PARENT)
+    public ApacheHttpClientConfig apacheHttp;
+
+    @ConfigGroup
+    public static class ApacheHttpClientConfig {
```

Comment:
If you don't use the APACHE client, how do you configure all these details? Proxy, TLS?

Reply:
There are no configuration options for URL connection other than socket timeout and connection timeout. But it's fine, as the lightweight url conn as sync client is fine for lambdas. For other app models like container talking to the dynamodb the apache sync client can be used that enables configurations for more connectivity aspects